packages/react-native/ReactCommon/react/runtime/platform/ios/ReactCommon/RCTHost.mm

```diff
@@ -254,7 +254,8 @@ - (RCTFabricSurface *)createSurfaceWithModuleName:(NSString *)moduleName
   surface.surfaceHandler.setDisplayMode(displayMode);
   [self _attachSurface:surface];
 
-  [_instance callFunctionOnBufferedRuntimeExecutor:[surface](facebook::jsi::Runtime &_) { [surface start]; }];
+  __weak RCTFabricSurface *weakSurface = surface;
+  [_instance callFunctionOnBufferedRuntimeExecutor:[weakSurface](facebook::jsi::Runtime &_) { [weakSurface start]; }];
```

Comment:
I'm not actually sure why this needs go through the JS thread, since `start` already dispatches async internally anyway and is threadsafe. It seems this was added in https://github.com/facebook/react-native/pull/45486 ?

Can we make this just `[surface start]` and remove the weak ref? It not, can you add a comment on why this requires synchronization on the JS thread?

Reply:
@javache Hi, I added the comment, `[surface start]` needs to wait for the main bundle to be loaded, so we dispatched it on the buffered runtime executor.

Reply:
I'm a bit surprised by that. ReactInstance should be able to deal with the queuing required to delay surface start until the bundle is ready.

Reply:
The surface start seems just schedule to the js thread :
https://github.com/facebook/react-native/blob/f4a17fbb93d6cdba29a340f0dde4d155af972439/packages/react-native/ReactCommon/react/renderer/uimanager/UIManager.cpp#L238-L242